dask/multiprocessing.py

```diff
@@ -4,9 +4,27 @@
 from .optimize import fuse, cull
 import multiprocessing
 import dill
+import cloudpickle
 import pickle
 from .async import get_async # TODO: get better get
 from .context import _globals
+from sys import getrecursionlimit, setrecursionlimit
+
+def _dumps(x):
+    try:
+        return cloudpickle.dumps(x, protocol=pickle.HIGHEST_PROTOCOL)
+    except:
+        try:
+            recursionlimit = getrecursionlimit()
+            setrecursionlimit(2*recursionlimit)
+            dumped = cloudpickle.dumps(x, protocol=pickle.HIGHEST_PROTOCOL)
+            setrecursionlimit(recursionlimit)
+            return dumped
+        except:
+            raise
```

Comment:
For what objects were we hitting a recursion limit?  Ideally nothing we do is this heavily nested.